tsinfer/formats.py

```diff
@@ -2309,6 +2309,11 @@ def __init__(self, path):
                     " sgkit dataset, indicating that all the genotypes are"
                     " unphased"
                 )
+        if np.unique(self.sites_position).shape[0] != self.sites_position.shape[0]:
```

Comment:
There's no need for a unique here, that's much more expensive than necessary. Can just say the np.diff must be greater than 0

Reply:
Oh, good catch. Yes, your suggested check should be fine. I have just updated the PR